modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/Utils.java

```diff
@@ -613,21 +613,8 @@ private static MosaicConfigurationBean loadMosaicProperties(
         //
         // load the properties file
         //
-        URL propsURL = sourceURL;
-        if (!sourceURL.toExternalForm().endsWith(".properties")) {
-            propsURL = URLs.changeUrlExt(sourceURL, "properties");
-            if (propsURL.getProtocol().equals("file")) {
-                final File sourceFile = URLs.urlToFile(propsURL);
-                if (!sourceFile.exists()) {
-                    if (LOGGER.isLoggable(Level.INFO)) {
-                        LOGGER.info("properties file doesn't exist");
-                    }
-                    return null;
-                }
-            }
-        }
-
-        final Properties properties = CoverageUtilities.loadPropertiesFromURL(propsURL);
+        final Properties properties = loadProperties(sourceURL);
+        if (properties == null) return null;
         if (properties == null) {
```

Comment:
two times the same check and return: 
if (properties == null) return null

Reply:
Good catch